src/sqlfluff/dialects/dialect_redshift_keywords.py

```diff
@@ -202,6 +202,7 @@
 ATTACH
 ATTRIBUTE
 ATTRIBUTES
+AUTO_INCREMENT
```

Comment:
Why are you adding in these keywords in this PR, there's no mention of motivation and they aren't used in the CREATE/ALTER USER/GROUP grammar or tests?

If they are intended please elaborate on why or provide examples of their usage in the CREATE/ALTER USER/GROUP unit tests.

If they're unrelated but you want to add them then make a separate PR 👍

Reply:
@jpy-git Myself and @tunetheweb had a discussion a while back about this on Slack. It seems that in order for Redshift to be able to parse ALTER USER or ALTER GROUP statements, those need to be added to the dialect.  Otherwise, I run into Keyword errors. I'm happy to add these in another PR though, but it seems to me that this is a blocker. Should I spin this off into another PR?

Reply:
My point is that they aren't referenced in any of the documentation for these statements:
- https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_USER.html
- https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_USER.html
- https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_GROUP.html
- https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_USER.html

If you were to remove them from this PR none of the existing unit tests would fail. That means if someone were to, for example, come along in the future and tidy up the list of redshift keywords then they could delete these because they're not in the [docs](https://docs.aws.amazon.com/redshift/latest/dg/r_pg_keywords.html) and they wouldn't know that they're valid/used since there's no tests that would fail if they get removed.

Basically fine with you including them in this PR if they actually get used in some of these statements and I understand that sometimes the docs can be incomplete. All you need to do is just put some simple unit tests in the relevant statement where they actually get used (same applies if you split this into a separate PR).

Reply:
These are not needed by RedShift dialect. They can be helpful due to the way SQLFluff works and dialect inheritance.

So basically if a grammar is not recognised, SQLFluff will eventually try to go through all the segments trying to make a match. This means it will also go through all the segment sin the ANSI dialect RedShift inherits from, and reach "ANSI" segments that don't exist in RedShift. Then rather than saying a segment is unparsable it will say a more cryptic message like "Grammar refers to 'Reference_autoincrementKeywordSegment' which was not found in the redshift dialect`.

So choices are:
1. Add all keywords in a dialect that it is inherited from (ANSI usually) as unreserved keywords. No real downside to this except, as @jpy-git points out someone might remove them thinking they are not needed. Maybe a comment will help, but haven't really done that for other dialects (though maybe should).
2. Override all non-supported ANSI segments in each dialect so unsupported Keywords are never tried to be used. This is probably the "right" answer but a bit unrealistic IMHO.
3. Do nothing and just deal with the somewhat cryptic error message (which should only happen in unparsable segments anyway).

Personally I prefer option 1. We shouldn't add unit tests for them, but maybe should add a comment.

Reply:
Found the slack conversation for this: https://sqlfluff.slack.com/archives/D02L1N7EATB/p1635725867011800

Reply:
Most dialects just inherit their keywords from ANSI for that reason to avoid this, so not sure why redshift explicitly clears them [here](https://github.com/sqlfluff/sqlfluff/blob/212432f5d9830566a23479769d770eaeb7774317/src/sqlfluff/dialects/dialect_redshift.py#L33) if we then have to then manually re-add the ANSI keywords into the redshift_keywords. Makes the Redshift keywords a bit messy if we want to implement option 2 in the future.
(@jpers36 recently implemented option 2 for T-SQL in #2170 so it's a good thing to aim for as a dialect gets more complete)

Reply:
Also these keywords are def not relevant to the title/description of this PR so should be separate with details on motivation (can't see the slack chat btw) in the future.

I'm not fussed either way so feel free to merge if you're happy 👍

Reply:
Inheriting keywords from ANSI can have other unintended side effects (especially for ANSI reserved keywords that shouldn’t be reserved in this dialect) but yes that is another option. The reserved list of keywords in ANSI is quite small anyway (and mostly reserved in other dialects too) and don’t _think_ there is a problem with a reserved keyword also being unreserved.

Anyway, I’ve added a comment to the “Are there any other side effects…” section of the initial comment to explain why these are being added despite only being tangentially related to this PR.